### PR TITLE
Version 0.0.2

### PR DESCRIPTION
**Pull request by nhayfield**
Most Done. Planning to merge it so we can do smaller prs

### CODE REVIEW

**Review by calebdoxsey — APPROVED** [review]

File: src/renderer/icons/ClosedFolder.tsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    width: '20px',
+  },
+}));
+
+const ClosedFolder = (): JSX.Element => {
+  const classes = useStyles();
+  return (
+    <SvgIcon viewBox="0 0 20 20" className={classes.iconStyle}>
+      <path
+        d="M8.33317 3.3335H3.33317C2.4165 3.3335 1.67484 4.0835 1.67484 5.00016L1.6665 15.0002C1.6665 15.9168 2.4165 16.6668 3.33317 16.6668H16.6665C17.5832 16.6668 18.3332 15.9168 18.3332 15.0002V6.66683C18.3332 5.75016 17.5832 5.00016 16.6665 5.00016H9.99984L8.33317 3.3335Z"
```
[the remark this answers is not in the document]
Comment (nhayfield) [reply src/renderer/icons/ClosedFolder.tsx:17]: no library. it is from the designer.

File: src/renderer/pages/ConnectForm.tsx [hunk 2]
```diff
@@ -0,0 +1,351 @@
...
+
+  const saveName = (value: string): void => {
+    setConnection({
+      ...connection,
+      ...{ name: value || undefined },
+    });
+  };
+
+  const saveTags = (arr: string[]): void => {
+    setTags(arr.map((tag) => formatTag(tag)));
```
[the remark this answers is not in the document]
Comment (nhayfield) [reply src/renderer/pages/ConnectForm.tsx:141]: yup

File: src/renderer/pages/ManageConnections.tsx [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+  },
+}));
+
+const ManageConnections = (): JSX.Element => {
+  const classes = useStyles();
+  const [folderNames, setFolderNames] = useState([] as string[]);
+  const [connections, setConnections] = useState([] as ListenerRecord[]);
+  const [statuses, setStatuses] = useState(
+    {} as { [key: string]: ListenerStatus }
+  );
```
[the remark this answers is not in the document]
Comment (nhayfield) [reply src/renderer/pages/ManageConnections.tsx:58]: yup had copy pasted from the generated code

File: src/renderer/pages/ManageConnections.tsx [hunk 4]
```diff
@@ -0,0 +1,304 @@
...
+  const [error, setError] = useState(null as ServiceError | null);
```
[the remark this answers is not in the document]
Comment (nhayfield) [reply src/renderer/pages/ManageConnections.tsx:59]: it complains unless i do this:  
`const [error, setError] = useState<ServiceError | null>(null);`

**Pull request merged**